Tests/GRPCProtobufTests/InputParserTests.swift

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCodeGen
+import GRPCProtobuf
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+import XCTest
+
+final class InputParserTests: XCTestCase {
+  func testParser() throws {
+    let parsedCodeGenRequest = try InputParser().parse(input: self.dummyFileDescriptor)
+    XCTAssertEqual(parsedCodeGenRequest.fileName, "helloworld.proto")
+    XCTAssertEqual(
+      parsedCodeGenRequest.leadingTrivia,
+      """
+      // DO NOT EDIT.
+      // swift-format-ignore-file
+      //
+      // Generated by the Swift generator plugin for the protocol buffer compiler.
+      // Source: helloworld.proto
+      //
+      // For information on using the generated types, please see the documentation:
+      //   https://github.com/apple/swift-protobuf/
+
+      // Copyright 2015 gRPC authors.\n//\n// Licensed under the Apache License, \
+      Version 2.0 (the \"License\");\n// you may not use this file except in \
+      compliance with the License.\n// You may obtain a copy of the License \
+      at\n//\n//     http://www.apache.org/licenses/LICENSE-2.0\n//\n// Unless required by \
+      applicable law or agreed to in writing, software\n// distributed under \
+      the License is distributed on an \"AS IS\" BASIS,\n// WITHOUT WARRANTIES \
+      OR CONDITIONS OF ANY KIND, either express or implied.\n// See the License \
+      for the specific language governing permissions and\n// limitations under the License.\n\n
+      """
+    )
+
+    let expectedMethod = CodeGenerationRequest.ServiceDescriptor.MethodDescriptor(
+      documentation: "/// Sends a greeting.\n",
+      name: CodeGenerationRequest.Name(
+        base: "SayHello",
+        generatedUpperCase: "SayHello",
+        generatedLowerCase: "sayHello"
+      ),
+      isInputStreaming: false,
+      isOutputStreaming: false,
+      inputType: "HelloRequest",
+      outputType: "HelloReply"
+    )
+    XCTAssertEqual(
+      parsedCodeGenRequest.services[0].methods[0].documentation,
+      "/// Sends a greeting.\n"
+    )
```

Comment:
Can we pull out the first service? I.e. `guard let service = request.services.first else { return XCTFail() }`

Reply:
Also we should validate the number of services and methods is as expected